conans/model/conan_file.py

```diff
@@ -132,6 +133,12 @@ def initialize(self, settings, env):
         self.options = create_options(self)
         self.requires = create_requirements(self)
         self.settings = create_settings(self, settings)
+
+        try:
```

Comment:
I think this check might be redundant:
- It is not possible to get cppstd defined if it doesn't come from the profile
- The profile was checked before being applied to conanfiles

If the problem is per-package settings, I think I'd try to implement that in the same place as the global cppstd is checked, to keep that logic together.

Reply:
Yes, this is about checking the _per-package settings_. 

I tried to do it (there are some ideas in [this PR](https://github.com/jgsogo/conan/pull/11) which I moved from this repo to mine), but there is a problem: **I need to constrain the settings for the package before checking them** and that action is performed inside this function, in the call to `self.settings = create_settings(self, settings)`.

I would really like to move this check to previous stages (this is executed once the graph starts to run), but it would require to move too many things not related to this PR, don't you think so?

Reply:
I don't get it. I will go ask you "live" LoL